libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.ts

```diff
@@ -266,15 +285,30 @@ export class SDSAutocompleteSearchComponent implements ControlValueAccessor {
       this.onArrowGroupUp();
     } else if (KeyHelper.is(KEYS.ENTER, event) && this.highlightedIndex >= 0) {
       if (this.configuration.isTagModeEnabled) {
-        const val = this.createFreeTextItem();
-        this.selectItem(val);
-      } else {
-        this.selectItem(this.highlightedItem);
+        this.error = this.checkValidaton(this.inputValue, this.field);
+        this.hasError = this.error === null ? false : true;
+        this.errorMessage = this.getErrorMessage(this.error);
+        if (!this.hasError) {
+          const val = this.createFreeTextItem();
+          this.selectItem(val);
+        } else {
+          this.selectItem(this.highlightedItem);
+        }
       }
     } else if (KeyHelper.is(KEYS.ENTER, event) && this.highlightedIndex < 0) {
       if (this.configuration.isFreeTextEnabled) {
-        const val = this.createFreeTextItem();
-        this.selectItem(val);
+        this.error = this.checkValidaton(this.inputValue, this.field);
+        this.hasError = this.error === null ? false : true;
+        this.errorMessage = this.getErrorMessage(this.error);
```

Comment:
Could you do a null check before calling getErrorMessage here? Line 325 will throw a null exception if error is null here.